File: src/main/java/com/ject/studytrip/studylog/domain/policy/StudyLogPolicy.java
package com.ject.studytrip.studylog.domain.policy;

import com.ject.studytrip.global.exception.CustomException;
import com.ject.studytrip.studylog.domain.error.StudyLogErrorCode;
import com.ject.studytrip.studylog.domain.model.StudyLog;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StudyLogPolicy {
    public static void validateNotDeleted(StudyLog studyLog) {
        if (studyLog.getDeletedAt() != null) {
            throw new CustomException(StudyLogErrorCode.STUDY_LOG_ALREADY_DELETED);
        }
    }
}

File: src/main/kotlin/com/ject/studytrip/studylog/domain/policy/StudyLogPolicy.kt
package com.ject.studytrip.studylog.domain.policy

import com.ject.studytrip.global.exception.CustomException
import com.ject.studytrip.studylog.domain.error.StudyLogErrorCode
import com.ject.studytrip.studylog.domain.model.StudyLog

object StudyLogPolicy {
    fun validateNotDeleted(studyLog: StudyLog) {
        if (studyLog.isDeleted()) {
            throw CustomException(StudyLogErrorCode.STUDY_LOG_ALREADY_DELETED)
        }
    }

    fun validateExistAll(
        foundStudyLogs: List<StudyLog>,
        requestedIds: List<Long>,
    ) {
        if (foundStudyLogs.size != requestedIds.size) {
            throw CustomException(StudyLogErrorCode.STUDY_LOG_NOT_FOUND)
        }
    }
}
